go/private/sdk.bzl

```diff
@@ -233,7 +237,7 @@ def _sdk_build_file(ctx, platform):
             "{goos}": goos,
             "{goarch}": goarch,
             "{exe}": ".exe" if goos == "windows" else "",
-            "{rules_go_repo_name}": Label("//go/private:BUILD.sdk.bazel").workspace_name,
```

Comment:
I think the logic behind this was that it would resolve to `io_bazel_rules_go` if you imported it as such, but if people started to import as `rules_go` they would get that. Does this force us to use `io_bazel_rules_go` going forward?

Reply:
This is a pattern that currently breaks with bzlmod: With repo mappings, which it uses, the value of `workspace_name` will be the *mapped* repo name, which can't be used to form a valid label. This is where https://github.com/bazelbuild/bazel/issues/15593 would help.

As a result, this line either needs to be replaced on master or via a patch in the BCR. I opted for the former as there are other blockers to a non-bzlmod repo name agnostic rules_go (see my other comment).